From ba9601882c11ade20d42e4f2d7ebc24f3cc6f07c Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 158/296] Native crash occurs.
 android.renderscript.Script#setVer()

Bug Details:
If null is specified for the argument dims of android.renderscript.Script#setVer(),
Native crash occurs without RSIllegalArgumentException being thrown

Modification:
Implemented to throw RSIllegalArgumentException if the argument dims of the setVer() method is null.
---
 rs/java/android/renderscript/Script.java | 4 ++++
 1 file changed, 4 insertions(+)
 mode change 100644 => 100755 rs/java/android/renderscript/Script.java

diff --git a/rs/java/android/renderscript/Script.java b/rs/java/android/renderscript/Script.java
old mode 100644
new mode 100755
index d0d9a114bba..5fc3c60847a
--- a/rs/java/android/renderscript/Script.java
+++ b/rs/java/android/renderscript/Script.java
@@ -450,6 +450,10 @@ public class Script extends BaseObj {
      *
      */
     public void setVar(int index, FieldPacker v, Element e, int[] dims) {
+        if (dims == null) {
+            throw new RSIllegalArgumentException(
+                "dims should to be non-null.");
+        }
         mRS.nScriptSetVarVE(getID(mRS), index, v.getData(), e.getID(mRS), dims);
     }
 
-- 
2.11.0

